### PR TITLE
chore(config): Test the rest of the global option merging

### LINKED ISSUE #15078: Fully validate configuration merging
In #15071, a bug was identified in how Vector merges the global `timezone` configuration setting. That individual bug was fixed, but we do not have tests to ensure that all other configuration merging happens properly. We should add a full set of unit tests to validate that configuration merging works for all of the fields of the global configuration, and for whatever parts of the component configurations make sense.

### PR DESCRIPTION
**Pull request by bruceg**
Closes #15078

### CODE REVIEW

File: lib/vector-core/src/config/global_options.rs [hunk 1]
```diff
@@ -258,4 +269,66 @@ mod tests {
...
+
+        assert_eq!(merge(None, None), Ok(None.into()));
+        assert_eq!(merge(Some(false), None), Ok(false.into()));
+        assert_eq!(merge(Some(true), None), Ok(true.into()));
+        assert_eq!(merge(None, Some(false)), Ok(false.into()));
+        assert_eq!(merge(None, Some(true)), Ok(true.into()));
+        assert_eq!(merge(Some(false), Some(false)), Ok(false.into()));
+        assert_eq!(merge(Some(true), Some(true)), Ok(true.into()));
+        assert_eq!(merge(Some(false), Some(true)), Ok(false.into()));
+        assert_eq!(merge(Some(true), Some(false)), Ok(true.into()));
```
Comment (bruceg) [inline lib/vector-core/src/config/global_options.rs:300]: I'm not sure this is right handling. This implies that the rule for conflicting settings for `acknowledgements` is first-one-wins, which is almost certainly not right. Should these result in a conflict or in one overriding the other?

Comment (tobz) [reply lib/vector-core/src/config/global_options.rs:300]: I'd say this should result in a conflict.

Configuring two distinct instances of a global acknowledgements setting, and trying to programmatically make a decision on which one has higher precedence, seems far too risky to me.

**Review by tobz — COMMENTED** [review]
This looks good overall, but left a few questions/comments.

File: lib/vector-core/src/config/global_options.rs (hunk 1 above)
Comment (bruceg) [reply lib/vector-core/src/config/global_options.rs:300]: Thanks, I'll update the handling to check this.

**Review by tobz — APPROVED** [review]
LGTM. 👍🏻

**Pull request merged**